File: frontend/src/client/types/audit.ts
/**
 * Audit Module Types
 *
 * Types for the audit module used to validate detected patterns against ATAS.
 */

/**
 * Request to generate Order Blocks audit report
 */
export interface AuditOrderBlocksRequest {
  symbol: string;
  timeframe: string;
  snapshot_time: string; // ISO 8601 datetime string
}

/**
 * Single Order Block for audit validation
 */
export interface OrderBlockAuditItem {
  ob_id: number;
  ob_type: string; // BULLISH OB, BEARISH OB, STRONG BULLISH OB, STRONG BEARISH OB
  formation_time_est: string; // Formatted: "Nov 6, 2025 05:20:00 EST"
  formation_time_utc: string; // For reference
  zone_low: number;
  zone_high: number;
  body_midpoint: number;
  range_midpoint: number;
  status: string; // ACTIVE, TESTED, BROKEN
  quality: string; // HIGH, MEDIUM, LOW
  impulse_move: number;
  impulse_direction: string; // UP, DOWN
  candle_direction: string; // BULLISH, BEARISH
  ob_open: number;
  ob_close: number;
  ob_volume: number;
}

/**
 * Response with Order Blocks audit report
 */
export interface AuditOrderBlocksResponse {
  report_markdown: string; // Markdown formatted report
  total_obs: number; // Total ACTIVE OBs at snapshot
  snapshot_time_est: string; // Formatted: "Nov 24, 2025 09:30:00 EST"
  snapshot_time_utc: string;
  symbol: string;
  timeframe: string;
  order_blocks: OrderBlockAuditItem[]; // Full data for programmatic access
}

File: backend/app/schemas/audit.py
"""
Audit Schemas

Pydantic schemas for audit report generation and validation.
"""
from datetime import datetime
from typing import List
from pydantic import BaseModel, Field


class AuditOrderBlocksRequest(BaseModel):
    """Request to generate Order Blocks audit report"""
    symbol: str = Field(..., example="NQZ5")
    timeframe: str = Field(..., example="5min")
    snapshot_time: datetime = Field(..., example="2025-11-24T14:30:00", description="Snapshot timestamp (UTC)")


class OrderBlockAuditItem(BaseModel):
    """Single Order Block for audit validation"""
    ob_id: int
    ob_type: str  # BULLISH OB, BEARISH OB, STRONG BULLISH OB, STRONG BEARISH OB
    formation_time_est: str  # Formatted: "Nov 6, 2025 05:20:00 EST"
    formation_time_utc: str  # For reference
    zone_low: float
    zone_high: float
    body_midpoint: float
    range_midpoint: float
    status: str  # ACTIVE, TESTED, BROKEN
    quality: str  # HIGH, MEDIUM, LOW
    impulse_move: float
    impulse_direction: str  # UP, DOWN
    candle_direction: str  # BULLISH, BEARISH
    ob_open: float
    ob_close: float
    ob_volume: float


class AuditOrderBlocksResponse(BaseModel):
    """Response with Order Blocks audit report"""
    report_markdown: str  # Markdown formatted report
    total_obs: int  # Total ACTIVE OBs at snapshot
    snapshot_time_est: str  # Formatted: "Nov 24, 2025 09:30:00 EST"
    snapshot_time_utc: str
    symbol: str
    timeframe: str
    order_blocks: List[OrderBlockAuditItem]  # Full data for programmatic access
